### PR TITLE
feat(input-stepper): add alignToStep property to align value with ste…

### PR DESCRIPTION
**Pull request by movrack**
Adding property alingToStep on input-stepper.
When property enabled, clicking on + or - button will verify the next step to reach and align on step multiple if user did enter custom value in between.

**example**
min 0
max 100
step 10
current value 55

click +  ==> 60 instead of 65
click - ==> 50 instead of 55

if min is not 0:

min 3
max 100
step 10
current value 55

click +  ==> 63 instead of 65
click - ==> 53 instead of 55

## What I did

1. Implement the property
2. Add test
3. Write documentation

I made multiple try with the changeset but I don't find how to do it correctly.

### CODE REVIEW

File: packages/ui/components/input-stepper/src/LionInputStepper.js [hunk 1]
```diff
@@ -266,7 +277,13 @@ export class LionInputStepper extends LocalizeMixin(LionInput) {
    */
   __increment() {
     const { step, min, max } = this.values;
-    const newValue = this.currentValue + step;
+    let newValue = this.currentValue + step;
+
+    if (this.alignToStep && (this.currentValue + min) % step !== 0) {
```
Comment (gerjanvangeest) [inline packages/ui/components/input-stepper/src/LionInputStepper.js:282]: This is native behaviour of the `<input type="number">`, so maybe we should not provide an option and just make this default. @ryubro @okadurin ?

Comment (ryubro) [reply packages/ui/components/input-stepper/src/LionInputStepper.js:282]: Yes, it makes better sense to me but we're introducing a break change. Maybe we should offer an option that is opposite to `alignToStep`, for those possible consumers who are using the current behavior as a feature?

Comment (gerjanvangeest) [reply packages/ui/components/input-stepper/src/LionInputStepper.js:282]: @movrack, we discussed it internally and we do not think this that big of a breaking change, we can defend this as a bug fix.
So can you remove the `alignToStep` property and make this the default behavoir?

**Comment by gerjanvangeest** [comment]
@movrack One more request, can you add a changeset `npm run changeset`, so the change ends up in our changelog?

**Review by gerjanvangeest — APPROVED** [review]

**Comment by movrack** [comment]
Looks like tests are not working but not related to my changes.
Any idea what it could be ?

**Comment by movrack** [comment]
Fixed and adding tests to try others limits/ edge cases with min below 0 for examples.

File: packages/ui/components/input-stepper/src/LionInputStepper.js [hunk 2]
```diff
@@ -265,8 +265,19 @@ export class LionInputStepper extends LocalizeMixin(LionInput) {
    * @private
    */
   __increment() {
-    const { step, min, max } = this.values;
-    const newValue = this.currentValue + step;
+    const { step, max } = this.values;
+    let { min } = this.values;
+    if (min === Infinity) {
```
Comment (gerjanvangeest) [inline packages/ui/components/input-stepper/src/LionInputStepper.js:270]: Can we align both functions?

I prefer the `const stepMin = min !== Infinity ? min : 0;` approach, thats more readable.
